test/runtime/lb.go

```diff
@@ -210,15 +210,57 @@ var _ = Describe("RuntimeLB", func() {
 
 		status = docker.ContainerExec(
 			"client",
-			"curl -s --fail --connect-timeout 4 http://2.2.2.2:80/public")
-		status.ExpectSuccess("L3 Proxy is not working IPv6")
+			"curl -s --fail --connect-timeout 4 http://[f00d::1:1]:80/public")
+		status.ExpectSuccess("L4 Proxy is not working IPv6")
 
 		By("L3 redirect to L4")
 		status = cilium.ServiceAdd(3, "2.2.2.2:0", []string{
 			fmt.Sprintf("%s:80", httpd1["IPv4"]),
 			fmt.Sprintf("%s:80", httpd2["IPv4"])}, 2)
 		status.ExpectFail("Service created with invalid data")
 	}, 500)
+
+	It("Service recovery on restart", func() {
+		containers("create")
+		cilium.EndpointWaitUntilReady()
+
+		httpd1, err := docker.ContainerInspectNet("httpd1")
+		Expect(err).Should(BeNil())
+
+		httpd2, err := docker.ContainerInspectNet("httpd2")
+		Expect(err).Should(BeNil())
+
+		status := cilium.ServiceAdd(1, "2.2.2.2:80", []string{
+			fmt.Sprintf("%s:80", httpd1["IPv4"]),
+			fmt.Sprintf("%s:80", httpd2["IPv4"])}, 2)
+		status.ExpectSuccess("L4 service can't be created")
+
+		status = docker.ContainerExec(
+			"client",
+			"curl -s --fail --connect-timeout 4 http://2.2.2.2:80/public")
+		status.ExpectSuccess("L4 Proxy is not working IPv4")
+
+		oldSvcIds, err := cilium.ServiceGetIds()
+		Expect(err).Should(BeNil())
+
+		res := cilium.Node.Exec("sudo systemctl restart cilium")
+		res.ExpectSuccess()
+
+		err = cilium.WaitUntilReady(100)
+		Expect(err).Should(BeNil())
+
+		eps := cilium.EndpointWaitUntilReady()
+		Expect(eps).Should(BeTrue())
+
+		svcIds, err := cilium.ServiceGetIds()
+		Expect(err).Should(BeNil())
+		Expect(len(svcIds)).Should(Equal(len(oldSvcIds)), "Service recovery does not match")
+
+		status = docker.ContainerExec(
+			"client",
+			"curl -s --fail --connect-timeout 4 http://2.2.2.2:80/public")
+		status.ExpectSuccess("LB is not working after restart")
+	})
```

Comment:
This does not fully port the functionality of `tests/06-lb.sh:test_svc_restore_functionality`. We need to be sure that the services and their associated metadata, not just the IDs, are equal. 
This includes frontends, backends, and IDs. For instance:
```
vagrant@cilium-master:~/go/src/github.com/cilium/cilium$ cilium service list -o json | python -m json.tool
[
    {
        "BackendAddresses": [
            "1 => 10.11.20.8:80",
            "2 => 10.11.94.160:80"
        ],
        "FrontendAddress": "2.2.2.2:80",
        "ID": 1
    }
]
```

Check that for the given ID that its associated FrontendAddress and BackendAddress are equal. Furthermore, there are no checks that the BPF LB maps are in sync which is done in `tests/06-lb.sh`. Please add that functionality.

Reply:
This still does not fully port the functionality in `tests/06-lb.sh:test_svc_restore_functionality`. It's necessary to check that the BPF maps match the Cilium service maps:
```
local SVC_IDS=$(cilium service list | tail -n+2 | awk '{ print $1 }')
  for id in $SVC_IDS; do
    check_cilium_bpf_maps_synced $id
  done
```